### DISCUSSION TITLE
Replacing Windows 11 explorer.exe with Windows 10 version

### DISCUSSION THREAD

**Discussion by kristibektashi** [valinet/ExplorerPatcher#361] [General]
If I replace the `C:\Windows\explorer.exe` with the version from build 19044 (Win10 21H2) or 20348 (Windows Server 2022) or 21390 (last Windows 10 branded Insider build) is it possible to bring back the Windows 10 Start menu or News and Interests?

**Comment by valinet** [comment]
I recently looked into this:
* `explorer.exe` from Windows 11 seems to have completely removed News and interests in favor of the Windows 11 widgets; the code was removed, or maybe it never existed on this branch; the taskbar right click menu still contains the submenu for "News and interests", but when the menu is built at runtime, that subitem is brutally removed using `DeleteMenu` without any check, so the chances of some component still enabling it to run under some circumstances are slim, I think
* `explorer.exe` from 19044 contained many references to News and interests in the code, at least part if not all of the functionality was implemented directly in the executable; I haven't tried running that on Windows 11, it probably does not work, but who knows... try, why not?
* `explorer.exe` from Server 2022 works on Windows 11; I haven't studied that, but I am pretty sure News and interests are not something to be found on a Server OS. Although it works on Windows 11, it has various glitches, I tested it last summer; for example, UWP apps do not show in the taskbar, which is pretty annoying.
* I do not recommend mixing and matching system files, as first of all it rarely has a chance of working, and second of all, it can lead to weird behaviors. But you can definitely test, of course. What's better to do then is to try to port the changes over from old versions, or things like that.
* Start Menu is not implemented in `explorer.exe`, but rather in `StartDocked.dll`; its entry point is the `StartMenuExperienceHost.exe` application
* The original Windows 10 Start menu code is still mostly intact in `StartDocked.dll` from recent Windows 11 builds
* Up until 22000.65 I think, there was a registry setting that allowed you to enable the old Start (`Start_ShowClassicMode` if I remember well). When it was removed, it was disregarded entirely, indeed. Now, the code, instead of checking that registry value, calls a function that checks whether the current Windows edition is PPI edition, or Windows Team, and only then it enables the Windows 10 Start menu.
* Indeed, Windows 11 Team still has the Windows 10 Start menu
* If you attach with a debugger when `StartMenuExperienceHost.exe` starts (attach to its parent, some `svchost.exe` instance, issue `.childdbg 1` in WinDbg, then kill `StartMenuExperienceHost.exe`; when the parent restarts the process, it will break at the entry point in WinDbg). So yeah, if you debug that, you can sset a breakpoint when that function check (`IsPPIEnabled` or something like that) is called and override the `rax` register and change it to `1`I think, which will fool it to think you are on Windows Team, and then the Windows 10 Start menu will load instead of the new menu.
* Unfortunately, it's not all roses: the menu has a couple of problems:
  * Right clicking anything crashes it
  * There is a noticeable delay when the menu closes, like 1 second, in which time the entire desktop is frozen (things work, just the image is frozen)
  * Live tiles do not seem to work anymore
* So yeah, it's pretty useless considering the above.
* Btw, you can grab a `StartDocked.dll` and `StartMenuExperienceHost.exe` from .51 or .65 or around there, replace the current files and you will get the Windows 10 Start menu without attaching via the debugger, only by setting that reg entry. Unfortunately, it still has the symptoms I described above, a sign that some external dependency is causing this. Which one, I don't really know.

That's basically my experience in this area, what I learned in the past couple of months regarding this topic.

Thanks

**Reply by kristibektashi** [reply]
What about `explorer.exe` from the last Windows 10 branded Insider build, [21390](https://betawiki.net/wiki/Windows_10_build_21390.1)? It had news and interests in it. Also it used the Cobalt codebase, the same one that Windows 11 uses. You can find the ISO on [UUPDump](https://uupdump.net) although you will probably need to set your date back to something like May 2021 and disable the network if you want to boot it.

**Reply by valinet** [reply]
Insider builds are worthless: they are signed in such a way that makes them timebombed - once the build expires, they can’t be used anymore, so not really a solution in the long run.

But, for the items queried in the original post, it’s not really needed:

* Start menu is not hosted by Explorer at all
* I have implemented a weather widget in the taskbar; it is coming in a future version, it’s now available in pre-releases, it works, but needs some polishing which I do not have time for at the moment.
